src/main/java/no/unit/nva/alma/AlmaRecordParser.java

```diff
@@ -0,0 +1,117 @@
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+
+public class AlmaRecordParser {
+
+    public static final String MARC_DATAFIELD_245 = "245";
+    public static final char MARC_SUBFIELD_A = 'a';
+    public static final char MARC_SUBFIELD_B = 'b';
+    public static final String BLANK = " ";
+    public static final String MARC_RECORD_XPATH = "//marc:record";
+    public static final String MARC_NAMESPACE = "http://www.loc.gov/MARC21/slim";
+    public static final String COLLECTION_ELEMENT = "collection";
+    public static final int FIRST_NODE = 0;
+
+
+
+    /**
+     * Parses a SRU-response to extract the title of an marcxml-record.
+     *
+     * @param inputStreamReader SRU-response
+     * @return simple json with <code>title</code>
+     * @throws IOException some stream reading went south
+     * @throws TransformerException some stream reading went south
+     * @throws SAXException some stream reading went south
+     * @throws ParserConfigurationException some stream reading went south
+     * @throws XPathExpressionException some stream reading went south
+     */
+    public Reference extractPublicationTitle(InputStreamReader inputStreamReader) throws IOException,
+            TransformerException, SAXException, ParserConfigurationException, XPathExpressionException {
+
+        Reference reference = new Reference();
+        try (InputStream inputStream = new ByteArrayInputStream(
+                CharStreams.toString(inputStreamReader).getBytes(StandardCharsets.UTF_8))) {
+
+            Record record = getFirstMarcRecord(inputStream);
+
+            if (isNull(record)) {
+                return reference;
+            }
+
+            DataField datafield = (DataField) record.getVariableField(MARC_DATAFIELD_245);
+            String title = datafield.getSubfields().stream()
+                    .filter(this::filterSubfields)
+                    .collect(Collectors.toMap(Subfield::getCode, Subfield::getData))
+                    .entrySet().stream().sorted(Map.Entry.comparingByKey())
+                    .map(Map.Entry::getValue).collect(Collectors.joining(BLANK));
+
+            reference.setTitle(title);
+            return reference;
+        }
+    }
+
+    private boolean filterSubfields(Subfield subfield) {
```

Comment:
Rename method to subfields we are filtering for?

Reply:
Done, filterSubfieldsAorB